src/WebJobs.Extensions.DurableTask/OutOfProcMiddleware.cs

```diff
@@ -198,6 +206,11 @@ await this.LifeCycleNotificationHelper.OrchestratorStartingAsync(
             OrchestratorExecutionResult orchestratorResult;
             if (functionResult.Succeeded)
             {
+                if (workerRequiresHistory)
+                {
+                    throw new SessionAbortedException("The worker has since ended the extended session and needs an orchestration history to execute the orchestration request.");

```

Comment:
Can we add tests for extended sessions in the .NET Isolated testing framework? We could add a test for the use case when this exception gets thrown.